src/ui/state/stakeholder.rs

```diff
@@ -146,3 +192,171 @@ impl From<StakeholderNetworkState> for Box<dyn State> {
         Box::new(s)
     }
 }
+
+#[derive(Debug)]
+pub struct StakeholderACKFundsState {
+    revaultd: Arc<RevaultD>,
+    warning: Watch<Error>,
+
+    balance: u64,
+    deposits: Vec<Deposit>,
+    view: StakeholderACKFundsView,
+}
+
+impl StakeholderACKFundsState {
+    pub fn new(revaultd: Arc<RevaultD>) -> Self {
+        StakeholderACKFundsState {
+            revaultd,
+            warning: Watch::None,
+            deposits: Vec::new(),
+            view: StakeholderACKFundsView::new(),
+            balance: 0,
+        }
+    }
+
+    fn update_deposits(&mut self, vaults: Vec<Vault>) -> Command<Message> {
+        self.calculate_balance(&vaults);
+        self.deposits = vaults.into_iter().map(|vlt| Deposit::new(vlt)).collect();
+        if let Some(Deposit::Pending { vault }) = self.deposits.first() {
+            return Command::perform(
+                get_revocation_txs(self.revaultd.clone(), vault.outpoint()),
+                |res| Message::Deposit(0, DepositMessage::RevocationTransactions(res)),
+            );
+        }
+        Command::none()
+    }
+
+    fn calculate_balance(&mut self, vaults: &Vec<Vault>) {
+        let mut balance: u64 = 0;
+        for vault in vaults {
+            match vault.status {
+                VaultStatus::Funded => {
+                    balance += vault.amount;
+                }
+                _ => {}
+            }
+        }
+
+        self.balance = balance;
```

Comment:
Aah good ol' imperative loop, nothing clearer ! :)
/me thinks about an intricate rustacean way of filter-map-then-sum, then hides